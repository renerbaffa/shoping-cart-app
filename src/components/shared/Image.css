.Image-loader {
  position: relative;
  margin: 6px 0px;
  display: flex;
  justify-content: center;
}