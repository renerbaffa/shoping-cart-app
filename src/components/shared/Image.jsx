import React from 'react';
import PropTypes from 'prop-types';

/*
  Image component is an adapter.
  Imagine that in the future we move to react-bootstrap or material-ui
  In this case we only need to replase default Image by the library Image
*/
const Image = ({ alt, src, ...other }) => (
  <img
    {...other}
    alt={alt}
    src={src}
  />
);

Image.propTypes = {
  src: PropTypes.string,
  alt: PropTypes.string,
}

Image.defaultProps = {
  src: '',
  alt: '',
}

export default Image;
