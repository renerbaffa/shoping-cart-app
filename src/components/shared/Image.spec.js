import React from 'react';
import renderer from 'react-test-renderer';

import Image from './Image';

describe('<Image />', () => {
  it('should render correctly', () => {
    const buttonSnapshot =
      renderer
        .create(
          <Image
            src="http://lorempixel.com/400/200/technics/"
            alt="alt"
          />
        )
        .toJSON();
    expect(buttonSnapshot).toMatchSnapshot();
  });
});
