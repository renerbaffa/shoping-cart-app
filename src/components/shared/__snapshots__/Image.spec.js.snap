// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render correctly 1`] = `
<img
  alt="alt"
  src="http://lorempixel.com/400/200/technics/"
/>
`;
